R/sql-context.R

```diff
@@ -0,0 +1,157 @@
+# The env for storing rolling_group_by methods
+polars_sql_context__methods <- new.env(parent = emptyenv())
+
+#' @export
+wrap.PlRSQLContext <- function(x) {
+  self <- new.env(parent = emptyenv())
+  self$`_ctxt` <- x
+
+  class(self) <- c("polars_sql_context", "polars_object")
+  self
+}
+
+#' Initialize a new `SQLContext`
+#'
+#' @param ... <[`dynamic-dots`][rlang::dyn-dots]> Elements that are known in the
+#' current `SQLContext`. It accepts any R object that can be converted to a
+#' LazyFrame via `as_polars_lf()`. All elements must be named.
+#'
+# TODO: add this param (see commit that dropped incomplete implementation in
+# https://github.com/eitsupi/neo-r-polars/pull/205)
+# @param .register_globals Register compatible objects found in the global
+# environment, automatically mapping their name to a table name. To register
+# other objects, pass them explicitly in `...`, or call the `execute_global`
+# class function.
+#'
+#' @return An object of class `"polars_sql_context"`
+#' @examples
+#' pl$SQLContext(mtcars = mtcars)
+#'
+#' pl$SQLContext(mtcars = mtcars, a = data.frame(x = 1))
+pl__SQLContext <- function(...) {
+  wrap({
+    self <- PlRSQLContext$new() |>
+      wrap()
+    self$register_many(...)
+    self
+  })
+}
+
+#' Register a single frame as a table, using the given name
+#'
+#' @param name Name of the table.
+#' @param frame Object to associate with this table name.
+#'
+#' @inherit pl__SQLContext return
+#' @examples
+#' df <- pl$DataFrame(x = 1)
+#' ctx <- pl$SQLContext()
+#' ctx$register("frame_data", df)$execute("SELECT * FROM frame_data")$collect()
+sql_context__register <- function(name, frame = NULL) {
+  wrap({
+    frame <- if (!is.null(frame)) {
+      as_polars_lf(frame)
+    } else {
+      pl$LazyFrame()
+    }
```

Comment:
I think this case separation is unnecessary since `as_polars_lf(NULL)` works.

Reply:
Ah nice, thanks.